docker/ubuntu-14.04-cpu-minimal/Dockerfile

```diff
@@ -0,0 +1,35 @@
+FROM ubuntu:14.04
+MAINTAINER Aaron Markham <aaronmarkham@fb.com>
+
+# caffe2 install with cpu support
+# 14.04 uses an older pip so using -U instead of --no-cache-dir
+
+########## REQUIRED DEPENDENCIES ################
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    build-essential \
+    cmake \
+    git \
+    libgoogle-glog-dev \
+    libprotobuf-dev \
+    python-pip \
+    protobuf-compiler \
+    python-dev \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN pip install -U --upgrade pip setuptools wheel && \
+    pip install -U future hypothesis numpy protobuf six
```

Comment:
FYI, the second `pip` command here won't actually use the upgraded pip.
https://lukeyeager.github.io/2016/11/29/upgrading-pip.html